docs/frontend/webpack-source-maps.md

```diff
@@ -6,8 +6,402 @@ which allows a developer to view / debug developer-friendly source code instead
+   rebuild the source map files, but there is some risk of version skew in the
+   build tools affecting the resulting bundle / source. The best practice is
+   to *store the maps that built the code in production*.
+3. Provide a simple development static server to serve the source maps at a
+   given port (in our example port `3000`) so that a project checkout can
+   provide the maps via a server.
+4. Ensure that the build process writes a correct `sourceMappingURL` comment
+   that corresponds to the correct locally hosted map server path.
+
+Then, when a developer needs to debug production code, they:
+
+1. Identify the version or git hash at issue.
+2. Download the project (git, npm, deploy artifact) at the proper version.
+3. Rebuild the map files if not already provided.
+4. Start the local static map server.
+
+After these steps, local source maps will be available for the production code.
+
+Advantages:
+
+* Easy implementation for existing infrastructure.
+* Don't need any specialized private map servers.
+
+Disadvantages:
+
+* Slow, tedious. Developers need to figure out the deployed version and check
+  out that version locally.
+* Rebuilding source maps risks correctness issues.
+
+**Option 2: Privately served maps**
+
+The more complicated and robust option is to have source mapping comments point
+to an internal server so that developers just have to "get on the network" and
+have access to source maps while the general public does not.
+
+For example, control comments in this scenario could look like:
+
+```js
+//# sourceMappingURL=http://vpn-only-server.com:3000/PATH/TO/app1.js.map
+//# sourceMappingURL=http://authenticated-server.com:3000/PATH/TO/app1.js.map
+```
+
+The application build process should then go through these steps:
+
+1. On building the application, publish the map files to a private server and
+   capture the published URL of each map file.
+2. Ensure that the build process writes a correct `sourceMappingURL` comment
+   that corresponds to the privately hosted map server path.
+
+Then, when a developer needs to debug production code, they:
+
+1. Gain access to the private maps server. This could be, logging in to
+   the VPN, authenticating to a publicly-accessible server, firing up an SSH
+   tunnel to a non-public server, etc.
+
+After these steps, privately hosted source maps will be available for the
+production code.
+
+Advantages:
+
+* Developers can just "go online" and magically have the correct maps.
+  No checking out versions, rebuilding, etc.
+
+Disadvantages:
+
+* Need to publish to private servers during build process.
+* Need to host / control private map servers.
+
+##### Source Map Example
+
+(Example build / dist code available at: [github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/webpack-source-maps](https://github.com/FormidableLabs/formidable-playbook/tree/master/examples/frontend/webpack-source-maps))
+
+Our Webpack configuration uses the
```

Comment:
Maybe we could start with the simple "Open Source" story first, then do this. And only after that get into the closed source business.